### PR TITLE
adds variable attr to top level object as well

### PR DESCRIPTION
**Pull request by johnricords**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+  target_id      = each.value.target_id
+  role_arn       = each.value.role_arn
+  event_bus_name = each.value.event_bus_name
+
+  dynamic "dead_letter_config" {
+    for_each = each.value.dead_letter_config != null ? [1] : []
+    content {
+      arn = each.value.dead_letter_config.arn
+    }
+  }
```
Comment (lorengordon) [inline main.tf:24]: use the dynamic block as the iteration object

```suggestion
  dynamic "dead_letter_config" {
    for_each = each.value.dead_letter_config != null ? [each.value.dead_letter_config] : []
    content {
      arn = dead_letter_config.value.arn
    }
  }
```

Comment (johnricords) [reply main.tf:24]: yeah, i had written it this way, but terraform refuses to recognize the object attribute in the content {} section

File: variables.tf [hunk 2]
```diff
@@ -21,6 +22,11 @@ variable "event_rule" {
         input_paths    = optional(map(string))
         input_template = string
       }))
+
+      sqs_target = optional(object({
+        message_group_id = optional(string)
```
Comment (lorengordon) [inline variables.tf:27]: as written, message_group_id is required if sqs_target is provided...

```suggestion
        message_group_id = string
```

File: main.tf [hunk 3]
```diff
@@ -21,4 +30,11 @@ resource "aws_cloudwatch_event_target" "this" {
       input_template = input_transformer.value.input_template
     }
   }
+
+  dynamic "sqs_target" {
+    for_each = each.value.sqs_target != null ? [1] : []
+    content {
+      message_group_id = each.value.sqs_target.message_group_id
+    }
+  }
```
Comment (lorengordon) [inline main.tf:39]: same here, use the dynamic block as the iteration object

```suggestion
  dynamic "sqs_target" {
    for_each = each.value.sqs_target != null ? [each.value.sqs_target] : []
    content {
      message_group_id = sqs_target.value.message_group_id
    }
  }
```

**Review by lorengordon — APPROVED** [review]
🚀

**Pull request merged**